Userland/Libraries/LibWeb/Fetch/Infrastructure/FetchAlgorithms.h

```diff
@@ -51,7 +51,13 @@ class FetchAlgorithms : public JS::Cell {
     virtual void visit_edges(JS::Cell::Visitor&) override;
 
 private:
-    explicit FetchAlgorithms(JS::VM&, Input);
+    explicit FetchAlgorithms(
+        JS::NonnullGCPtr<JS::HeapFunction<void(u64)>> process_request_body_chunk_length,
+        JS::NonnullGCPtr<JS::HeapFunction<void()>> process_request_end_of_body,
+        JS::NonnullGCPtr<JS::HeapFunction<void(JS::NonnullGCPtr<Infrastructure::Response>)>> process_early_hints_response,
+        JS::NonnullGCPtr<JS::HeapFunction<void(JS::NonnullGCPtr<Infrastructure::Response>)>> process_response,
+        JS::NonnullGCPtr<JS::HeapFunction<void(JS::NonnullGCPtr<Infrastructure::Response>)>> process_response_end_of_body,
+        JS::NonnullGCPtr<JS::HeapFunction<void(JS::NonnullGCPtr<Infrastructure::Response>, BodyBytes)>> process_response_consume_body);
```

Comment:
Can we alias all these long function types instead of repeating them N times?

I think we can do something like:
```c++
using ProcessResponseConsumeBodyHeapFunction = JS::NonnullGCPtr<JS::HeapFunction<ProcessResponseConsumeBodyFunction::FunctionType>>;
```

Also side note, this constructor being `explicit` still doesn't do anything, so that could be dropped

Reply:
(Or just use e.g. `ProcessResponseConsumeBodyFunction::FunctionType` inline in the constructor parameters / class members so we at least don't have to repeat the  underlying function type)